core/server/models/base/when.js

```diff
@@ -0,0 +1,59 @@
+var config = require('../../config'),
```

Comment:
i saw that david also created a similar file called `events`, would create a task and merge them together